doc/kernel/memory_management/heap.rst

```diff
@@ -165,6 +165,16 @@ the kernel not to define the heap memory pool object. The maximum size is limite
 by the amount of available memory in the system. The project build will fail in
 the link stage if the size specified can not be supported.
 
+In addition, each subsystem (board, driver, library, etc) can set a custom
+requirement by defining a Kconfig option with the prefix
+HEAP_MEM_POOL_ADD_SIZE. If multiple subsystems specify custom values, the
+sum of these will be used as the minimum requirement. If the application
+tries to set a value that's less than the minimum value, this will be ignored
+and the minimum value will be used instead. To force a smaller than minimum
+value to be used, the application may enable the
+:kconfig:option:`CONFIG_HEAP_MEM_POOL_IGNORE_MIN` option, however this is
+discouraged as it may result in a non-working application image.
```

Comment:
Not sure I would use these words, because the value **is** likely to be wrong, it should be the worst case scenario, as discussed yesterday people will need to optimise their end applications manually, would be good to capture that knowledge here.

Reply:
I rephrased this.